src/shared.ts

```diff
@@ -185,15 +185,18 @@ export function maintainScrollVisibility(
 	scrollParent: HTMLElement
 ) {
 	const { offsetHeight, offsetTop } = activeElement
+	const { bottom, top } = activeElement.getBoundingClientRect()
 	const { offsetHeight: parentOffsetHeight, scrollTop } = scrollParent
 
 	const isAbove = offsetTop < scrollTop
 	const isBelow = offsetTop + offsetHeight > scrollTop + parentOffsetHeight
+	const isActiveElementLowerThanBottomOfScreen = bottom > document.documentElement.clientHeight
+	const isActiveElementHigherThanTopOfScreen = top < 0
 
-	if (isAbove) {
-		scrollParent.scrollTo(0, offsetTop)
-	} else if (isBelow) {
-		scrollParent.scrollTo(0, offsetTop - parentOffsetHeight + offsetHeight)
+	if (isAbove || isActiveElementHigherThanTopOfScreen) {
+		activeElement.scrollIntoView(true)
```

Comment:
Awesome stuff, @ikniaziuk! Thank you!

It's so great to see the scrolling w/ keyboard navigation working so well: https://deploy-preview-82--vue-accessible-selects.netlify.app/?path=/docs/selectsingle--primary

My only request: is there a way to do this "scroll into view" that is only relative to the parent element? When I was playing around on the deploy preview, whenever my whole page was scrolled & the select was near the top or the bottom, then navigating with the keyboard caused the entire page to scroll at some points, which got to be pretty distracting. I think in those cases, if possible, we'd only want to scroll the interior of the dropdown, to the correct position?

Demo:
![Screen+Recording+2021-03-19+at+02 45 25+AM](https://user-images.githubusercontent.com/14101491/111741868-9058b200-885d-11eb-81ad-472d46d09bb7.gif)

Reply:
Oh I see. I didn't check this. We have cases when list of options outside of the screen 
![image](https://user-images.githubusercontent.com/20156342/111751041-0f250d80-88a5-11eb-96e3-2052d8f9c62d.png)
And it would be great if parent scroll move to when user press f.e. arrow down. What do you think @jackkoppa  but this should happen only when list outside of the screen. I will fix the strange behaviour above

Reply:
@jackkoppa . I changed the behaviour. Now it uses two types of scroll - `scrollIntoView` when activeElement outside of th screen and `scrollTop` when we need scroll only parent element.